src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -1844,8 +1844,8 @@ fn pat_constructors<'tcx>(
                 smallvec![FixedLenSlice(pat_len)]
             }
         }
-        PatKind::Or { .. } => {
-            bug!("support for or-patterns has not been fully implemented yet.");
+        PatKind::Or { ref pats } => {
+            pats.iter().flat_map(|pat| pat_constructors(tcx, param_env, pat, pcx)).collect()
```

Comment:
If we're adding some kind of handling for or-patterns here, then we should also add some tests (or leave these changes for a later PR).

Reply:
Should I revert those changes then ?

Reply:
Yeah, I'd prefer to revert them and include them in a follow up PR.